### PR TITLE
Better query groups snippets

### PR DESCRIPTION
**Pull request by agourlay**
This PR introduces better test snippets for the query group API.

I added missing conversions to keep things lightweight.

### CODE REVIEW

File: src/qdrant_client/builders/query.rs [hunk 1]
```diff
@@ -14,6 +15,16 @@ impl QueryPointsBuilder {
...
 }
 
+impl QueryPointGroupsBuilder {
+    pub fn add_prefetch(mut self, prefetch_query: impl Into<PrefetchQuery>) -> Self {
+        match self.prefetch {
+            Some(ref mut prefetch) => prefetch.push(prefetch_query.into()),
+            None => self.prefetch = Some(vec![prefetch_query.into()]),
+        }
+        self
+    }
```
Comment (timvisee) [inline src/qdrant_client/builders/query.rs:25]: Though this is fine, we could simplify to something like [this](https://doc.rust-lang.org/std/option/enum.Option.html#method.get_or_insert_with) (untested):

```suggestion
    pub fn add_prefetch(mut self, prefetch_query: impl Into<PrefetchQuery>) -> Self {
        self.prefetch.get_or_insert_with(Vec::new).push(prefetch_query.into());
        self
    }
```

**Review by timvisee — APPROVED** [review]
I like these examples!

Did you not update the docs example on purpose, because this new snippet is rather large?

File: src/qdrant_client/builders/query.rs (hunk 1 above)
Comment (agourlay) [reply src/qdrant_client/builders/query.rs]: Thanks, applied it to the second builder as well https://github.com/qdrant/rust-client/pull/181/commits/25277d6f16319435abb5f553f23a4834f66f3b69

**Comment by agourlay** [comment]
> Did you not update the docs example on purpose, because this new snippet is rather large?

I actually forgot but it is true that is quite large.
Decided to make the example a bit more useful by advertising the `add_prefetch` builder https://github.com/qdrant/rust-client/pull/181/commits/46deef37b6bc36398432c574b2ecc5886babb012

**Pull request merged**